### DISCUSSION TITLE
친구신청 & 친구삭제 및 차단 알림 websocket 사용 유무

### DISCUSSION THREAD

**Discussion by nyeoni** [GhostPangPang/GhostPong#46] [General]
친구 신청 수락 소켓
친구 삭제 및 차단 소켓
할 것인지 여부에 대해 이야기해봐요!

**Comment by aLVINlEE9** [comment]
Web Socket에 모든 정보를 담아서 한 번에 처리할 것인지 
Web Socket에 flag로 알려주고, 클라이언트가 다시 HTTP Request를 해서 Response로 받을 것인지 정해봐야할것 같습니다

**Comment by jis-kim** [comment]
230411 회의에서 resolve 되었습니다.

## 친구 신청 수락

- 친구 목록 페이지에서 친구 신청 수락이 일어난 경우
    → Web Socket 통신
- 이유 : 새로운 메세지 방을 만들어주기 위함 (원활한 메세지 통신을 위해)
    1. 메세지방을 보고있다면
        1. web socket 에서 메세지에 대한 정보 뿐만 아니라, 그 새로운 친구에 대한 info(image, …) 가 필요해서 따로 소켓 통신 해주어야 함

### 승수의 의견 (Web Socket에 담을 정보)

Web Socket에 모든 정보를 담아서 한 번에 처리할 것인지
Web Socket에 flag로 알려주고, 클라이언트가 다시 HTTP Request를 해서 Response로 받을 것인지
⇒  “친구 생겼음”을 Web Socket 으로 flag 형식으로 알리고 친구 info를 get할 때 에러 핸들링이 필요하기 때문에 REST API 사용!
## 친구 끊기 / 차단
- 친구와 메세지 보내는 도중에 친구가 끊긴 경우
    → Web Socket 통신 X
친구가 끊어졌다고 뷰를 실시간으로 달라지게 하는 것은 사용자 경험에 좋지 않다고 판단
서버에서만 친구가 아닐 시 에러 처리 해두고, 일단 뷰는 유지하기로 결정
